lib/beaker/hypervisor/openstack.rb

```diff
@@ -115,13 +127,26 @@ def provision
         # Associate a public IP to the server
         # Create if there are no floating ips available
         #
-        ip = @compute_client.addresses.find { |ip| ip.instance_id.nil? }
-        if ip.nil?
-          @logger.debug "Creating IP for #{host.name} (#{host[:vmhostname]})"
-          ip = @compute_client.addresses.create
+        # Do we already have an address?
```

Comment:
Was this code here before?  This feels like a ride along that should be split out into a separate commit or PR.